### PR TITLE
Get assets working

### PR DESCRIPTION
**Pull request by jakeboxer**
This branch brings in the [sprockets](https://github.com/sstephenson/sprockets) gem and makes Hyperloop apps serve assets from `app/assets/images`, `app/assets/javascripts`, and `app/assets/stylesheets`, just like a Rails app would. It even does Sass and CoffeeScript!

### CODE REVIEW

File: spec/fixtures/assets/app/assets/javascripts/my-scripts.coffee [hunk 1]
```diff
@@ -0,0 +1 @@
+alert("such javascript wow");
```
Comment (jessicard) [inline spec/fixtures/assets/app/assets/javascripts/my-scripts.coffee:1]: Why are you using parens and semicolons for coffeescript?

File: lib/hyperloop/application.rb [hunk 2]
```diff
@@ -58,6 +66,32 @@ def call(env)
...
+    #
+    # Returns a Sprockets::Environment.
+    def assets
+      @assets ||= Sprockets::Environment.new do |env|
+        env.append_path(@root + "/app/assets/images")
+        env.append_path(@root + "/app/assets/javascripts")
+        env.append_path(@root + "/app/assets/stylesheets")
+
+        # compress everything in production
+        if ENV["RACK_ENV"] == "production"
```
Comment (jessicard) [inline lib/hyperloop/application.rb:88]: This is dumb, but is there a specific reason why you're mixing " and ' ?

File: spec/application_spec.rb [hunk 3]
```diff
@@ -92,4 +93,64 @@
...
+
+    it 'responds successfully to a request for the javascript app bundle' do
+      response = @request.get('/assets/app.js')
+
+      expect(response).to be_ok
+      expect(response.content_type).to eql('application/javascript')
+      expect(response.body).to match(/alert\("such javascript wow"\);/)
+    end
+
+    it "responds successfully to a request for a gif" do
```
Comment (jessicard) [inline spec/application_spec.rb:126]: Same question about mixing " and ' down here.

File: spec/application_spec.rb [hunk 4]
```diff
@@ -92,4 +93,64 @@
...
+      expect(response.content_type).to eql('image/gif')
+      expect(Digest::SHA1.hexdigest(response.body)).to eql('bcbc6e6fc1eb77b2ca676e17425df93a56495bb2')
+    end
+
+    it "responds successfully to a request for a jpg" do
+      response = @request.get("/assets/my-jpg.jpg")
+
+      expect(response).to be_ok
+      expect(response.content_type).to eql('image/jpeg')
+      expect(Digest::SHA1.hexdigest(response.body)).to eql('ae6a26b513d6648446da1395ee73e9cf32c8c668')
```
Comment (jessicard) [inline spec/application_spec.rb:139]: ew what the fuck lol

**Comment by jessicard** [comment]
welp thats all ive got, looks good, surprised its that simple

File: spec/fixtures/assets/app/assets/javascripts/my-scripts.coffee (hunk 1 above)
Comment (jakeboxer) [inline spec/fixtures/assets/app/assets/javascripts/my-scripts.coffee:1]: Cuz fuck skipping parens, but yeah I should probably kill the semicolon.

Comment (jessicard) [inline spec/fixtures/assets/app/assets/javascripts/my-scripts.coffee:1]: mr I WANNA USE COFFEESCRIPT BUT NOT REALLY

File: lib/hyperloop/application.rb (hunk 2 above)
Comment (jakeboxer) [inline lib/hyperloop/application.rb:88]: Nope, I've been a little inconsistent all over the place. I'll fix it in a separate pull so it doesn't clutter this one up.

File: spec/application_spec.rb (hunk 4 above)
Comment (jakeboxer) [inline spec/application_spec.rb:139]: Couldn't think of a better way to test the body contents :cry: any suggestions?

Comment (jessicard) [inline spec/application_spec.rb:139]: nope, just wanted to complain

Comment (jakeboxer) [inline spec/application_spec.rb:139]: :+1:

**Pull request merged**